### DISCUSSION TITLE
Time-based AND GIT-based job trigger

### DISCUSSION THREAD

**Discussion by gaigaslab-operations** [concourse/concourse#8456] [Help & Support]
I may be missing something, but I'm trying to trigger a job when BOTH a time resource is TRUE (i.e. within a time window) AND a git resource has a NEW commit.  

Whenever I configure both resources to be `trigger: true`, EITHER one can trigger the job, so we get job triggers OUT of the window or continual triggers within the window even though there is no new commit in github.

Am I missing something?

**Comment by marco-m** [comment]
Unfortunately this is how triggering work: it is always OR, never AND.

I see two possibilities:

1. You could modify the git resource "check" script to return a new version only within the time window you want.

2. You could "decouple" the pipeline with a "signaling device". This has the "advantage" that can be done without messing with resources. 
- Remove the time resource.
- Configure the git resource to trigger a job. That job will look at the current time and, if within the window, will do something to signal. This something can be a S3 put or even a git put to another repo.
- Finally the "real" pipeline: start with a job that has two inputs: one the real git repo, but set as non-triggering.
- The other input, set as triggering, will be your signal: either s3 or the other git repo.

EDIT 
Mhh, I take back option 2, I don't think it will work.

**Comment by gaigaslab-operations** [comment]
Thank you very much! This is exactly what I was thinking and it is good to
have backup as a known pattern.
